src/Microsoft.AspNetCore.Razor.Language/Intermediate/MalformedDirectiveIRNode.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright(c) .NET Foundation.All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.AspNetCore.Razor.Language.Intermediate
+{
+    public sealed class MalformedDirectiveIRNode : RazorIRNode
+    {
+        public MalformedDirectiveIRNode(DirectiveIRNode directive)
+        {
+            Annotations = new DefaultItemCollection();
+
+            foreach (var annotation in directive.Annotations)
+            {
+                Annotations.Add(annotation);
+            }
+
+            Diagnostics = new DefaultDiagnosticCollection();
+
+            for (var i = 0; i < directive.Diagnostics.Count; i++)
+            {
+                Diagnostics.Add(directive.Diagnostics[i]);
+            }
+
+            for (var i = 0; i < directive.Children.Count; i++)
+            {
+                Children.Add(directive.Children[i]);
+            }
+
+            Source = directive.Source;
+            Name = directive.Name;
+            Descriptor = directive.Descriptor;
```

Comment:
If you just created these during lowering instead of having a pass do it, none of this silliness would be required